applications/StructuralMechanicsApplication/custom_constitutive/plasticity_isotropic_kinematic_j2.cpp

```diff
@@ -0,0 +1,525 @@
+//************************************************************************************
+
+PlasticityIsotropicKinematicJ2::PlasticityIsotropicKinematicJ2()
+    : ConstitutiveLaw()
+{
+}
+
+//********************************COPY CONSTRUCTOR************************************
+//************************************************************************************
+
+PlasticityIsotropicKinematicJ2::PlasticityIsotropicKinematicJ2(const PlasticityIsotropicKinematicJ2 &rOther)
+    : ConstitutiveLaw(rOther)
+{
+}
+
+//********************************CLONE***********************************************
+//************************************************************************************
+
+ConstitutiveLaw::Pointer PlasticityIsotropicKinematicJ2::Clone() const
+{
+    PlasticityIsotropicKinematicJ2::Pointer p_clone(new PlasticityIsotropicKinematicJ2(*this));
+    return p_clone;
+}
+
+//********************************DESTRUCTOR******************************************
+//************************************************************************************
+
+PlasticityIsotropicKinematicJ2::~PlasticityIsotropicKinematicJ2()
+{
+}
+
+//************************************************************************************
+//************************************************************************************
+
+bool PlasticityIsotropicKinematicJ2::Has(const Variable<bool>& rThisVariable)
+{
+    if(rThisVariable == INELASTIC_FLAG){
+        return true;
+    }
+    return false;
+}
+
+//************************************************************************************
+//************************************************************************************
+
+bool& PlasticityIsotropicKinematicJ2::GetValue(
+    const Variable<bool>& rThisVariable,
+    bool& rValue
+    )
+{
+    if(rThisVariable == INELASTIC_FLAG){
+        rValue = mPlasticEvolution;
+    }
+
+    return rValue;
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void PlasticityIsotropicKinematicJ2::InitializeMaterial(
+    const Properties& rMaterialProperties,
+    const GeometryType& rElementGeometry,
+    const Vector& rShapeFunctionsValues
+    )
+{
+    mPlasticStrainLast = ZeroVector(this->GetStrainSize());
+    mPlasticStrain = ZeroVector(this->GetStrainSize());
+    mEquivalentPlasticStrainLast = 0.0;
+    mEquivalentPlasticStrain = 0.0;
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void PlasticityIsotropicKinematicJ2::FinalizeSolutionStep(
+    const Properties& rMaterialProperties,
+    const GeometryType& rElementGeometry,
+    const Vector& rShapeFunctionsValues,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    mPlasticStrainLast = mPlasticStrain;
```

Comment:
This function is deprecated, please don’t use it in new code

Reply:
Ok. I will remove them...
New functions are already implemented.